src/Control/Effect/Fail.hs

```diff
@@ -18,11 +18,8 @@ import Data.Coerce
 import Prelude hiding (fail)
 
 newtype Fail (m :: * -> *) k = Fail String
-  deriving (Functor)
-
-instance HFunctor Fail where
-  hmap _ = coerce
-  {-# INLINE hmap #-}
+  deriving stock Functor
+  deriving anyclass HFunctor
```

Comment:
I guess we could also have done `instance HFunctor Fail` w/o a `where` clause, but I must admit I like the `DerivingStrategies` here.